### PR TITLE
expose prognosis type

### PR DESCRIPTION
**Pull request by HybridFox**
Expose the progress types on stopovers so actual progress can be checken

### CODE REVIEW

**Comment by vkrause** [comment]
Given the values in this property ("PROGNOSED", "CALCULATED"), I'm wondering whether "Prog" is really meaning "progress" here, or rather "prognosis"?

**Comment by HybridFox** [comment]
> 

That seems very plausible! I will update it

File: parse/journey-leg.js [hunk 1]
```diff
@@ -140,6 +140,14 @@ const parseJourneyLeg = (ctx, pt, date) => { // pt = raw leg
 				// todo: parse & use `code: EXTERNAL_ID` remarks?
 			}
 
+			// ensure that a station that has been passed has the previous ones as passed as well
+			for (let i = res.stopovers.length - 1; i >= 0; i--) {
+				if (res.stopovers[i].hasArrived === true && res.stopovers[i - 1]) {
```
Comment (derhuerst) [inline parse/journey-leg.js:145]: Where does the `hasArrived` flag come from? It's not implemented yet, right?

File: parse/journey-leg.js [hunk 2]
```diff
@@ -140,6 +140,14 @@ const parseJourneyLeg = (ctx, pt, date) => { // pt = raw leg
 				// todo: parse & use `code: EXTERNAL_ID` remarks?
 			}
 
+			// ensure that a station that has been passed has the previous ones as passed as well
+			for (let i = res.stopovers.length - 1; i >= 0; i--) {
```
[the remark this answers is not in the document]
Comment (HybridFox) [reply parse/journey-leg.js:144]: It seems i have committed too much with my last fix, let me undo these changes

**Comment by derhuerst** [comment]
Thanks! In 7c68f962 & 95af0a01, I have expanded parsing of `*progType` to `parseJourneyLeg`, `parseArrival` & `parseDeparture`, and adapted the tests.

**Comment by derhuerst** [comment]
The changes have been published as `hafas-client@5.25.0`. 🎉

**Pull request closed without merging**